plumbum/machines/ssh_machine.py

```diff
@@ -319,8 +327,11 @@ def upload(self, src, dst):
         if IS_WIN32:
             src = self._translate_drive_letter(src)
             dst = self._translate_drive_letter(dst)
-        self._scp_command(src, f"{self._fqhost}:{shquote(dst)}")
+        self._scp_command(src, f"{self._get_wrapped_fqhost()}:{shquote(dst)}")
 
+    def _get_wrapped_fqhost(self):
+        host = f'[{self.host}]' if is_ipv6_address(self.host) and not self.host.startswith('[') else self.host
```

Comment:
testing `self.host.startswith` is cheaper than `is_ipv6_address`, so it should be performed first.